app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java

```diff
@@ -0,0 +1,135 @@
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB5;
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB6;
+
+/**
+ * AboutMac class.
+ * @author Aayush
+ * @version 1.0.
+ */
+
+public class AboutMACActivity extends AppCompatActivity {
+
+    /**
+     * About Mac Text.
+     */
+    @BindView(R.id.about_mac_text)
+    TextView textView;
+
+    /**
+     * About Mac Image.
+     */
+    @BindView(R.id.about_mac_image)
+    ImageView imageView;
+
+    /**
+     * About Mac Toolbar.
+     */
+    @BindView(R.id.toolbar_about_mac)
+    Toolbar toolbar;
+
+    /**
+     * Database Reference.
+     */
+    private DatabaseReference databaseReference;
+
+    /**
+     * Vale event listener.
+     */
+    private ValueEventListener listener;
+
+    /**
+     * Sharedpreferences.
+     */
+    private SharedPreferences sharedPreferences;
+
+    /**
+     * String text.
+     */
+    private String text;
+
+    /**
+     * String url.
+     */
+    private String url;
+
+    /**
+     * Tag for about mac activity.
+     */
+    private final String TAG = AHC.TAG + ".activities." + getClass().getSimpleName();
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_about_mac);
+        ButterKnife.bind(this);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        sharedPreferences = getSharedPreferences(AHC.USER_PREFERENCES, Context.MODE_PRIVATE);
+        text = sharedPreferences.getString(AHC.TEXT_ABOUT_MAC, "");
+        url = sharedPreferences.getString(AHC.IMAGEURL_ABOUT_MAC, "");
+        if ("".equals(text)) {
+            textView.setText(TIAB1 + TIAB2 + TIAB3 + TIAB4 + TIAB5 + TIAB6);
+        } else {
+            textView.setText(text);
+        }
+        databaseReference = FirebaseDatabase.getInstance().getReference().child(AHC.FDR_ABOUT_MAC);
+        listener = databaseReference.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(final DataSnapshot dataSnapshot) {
+                text = dataSnapshot.child(AHC.TEXT_ABOUT_MAC).getValue(String.class);
+                url = dataSnapshot.child(AHC.IMAGEURL_ABOUT_MAC).getValue(String.class);
+                textView.setText(text);
+                Glide.with(getApplicationContext()).load(url)
+                        .thumbnail(AHC.MAGICNUMBER_ABOUT_MAC)
+                        .into(imageView);
+                final SharedPreferences.Editor editor = sharedPreferences.edit();
+                editor.putString(AHC.TEXT_ABOUT_MAC, text);
+                editor.putString(AHC.IMAGEURL_ABOUT_MAC, url);
+                editor.apply();
+            }
+
+            @Override
+            public void onCancelled(final DatabaseError databaseError) {
+                Log.e(TAG, "ERROR");
```

Comment:
The error message is extremely vague.
check for a method in the `databaseError` obj that gives more details
Also use the printStackTrace method